lib/guest_installation_and_configuration_base.pm

```diff
@@ -850,42 +868,234 @@ sub config_guest_network_bridge {
+  _zone => $_zone, _bridge_type => $_bridge_type, _bridge_ports => $_bridge_ports, 
+  _bridge_stp => $_bridge_stp, _bridge_forwarddelay => $_bridge_forwarddelay])
+
+Write network device settings to conventional /etc/sysconfig/network/ifcfg-* or 
+/etc/NetworkManager/system-connections/*.nmconnection depends on whether system
+network is managed by NetworkManager or not. The supported arguments are listed
+out as below:
+$_ipaddr: IP address/mask length pair of the interface
+$_name: Identifier of the interface
+$_bootproto: DHCP automatic or manual configuration, 'static', 'dhcp' or 'none'
+$_startmode: Auto start up or connection: 'auto', 'manual' or 'off'
+$_zone: The trust level of this network connection
+$_bridge_type: 'master' or 'slave' to indicate master or slave interface
+$_bridge_port: Specify interface's master or slave interface name
+$_bridge_stp: 'on' or 'off' to turn stp on or off
+$_bridge_forwarddelay: The stp forwarding delay in seconds
+If $_ipaddr given is empty, it means there is no associated specific ip address 
+to this interface which might be attached to another bridge interface or will not 
+be assigned one ip address from dhcp, so set $_ipaddr to '0.0.0.0'.If $_ipaddr 
+given is non-empty but not in ip address format,for example, 'host-default',it 
+means the interface will not use a ip address from pre-defined subnet and will 
+automically accept dhcp ip address from public facing host network.
+
+=cut
+
 sub write_guest_network_bridge_device_config {
-    my ($self, $_ipaddr, $_name, $_bootproto, $_startmode, $_zone, $_bridge, $_bridge_ports, $_bridge_stp, $_bridge_forwarddelay) = @_;
-
-    $self->reveal_myself;
-    $_ipaddr //= '0.0.0.0';
-    $_name //= '';
-    $_bootproto //= 'dhcp';
-    $_startmode //= 'auto';
-    $_zone //= '';
-    $_bridge //= '';
-    $_bridge_ports //= '';
-    $_bridge_stp //= 'off';
-    $_bridge_forwarddelay //= '15';
-    script_run("cp /etc/sysconfig/network/ifcfg-$_name /etc/sysconfig/network/backup-ifcfg-$_name");
-    script_run("cp /etc/sysconfig/network/backup-ifcfg-$_name $common_log_folder");
-    my $_bridge_device_config_file = '/etc/sysconfig/network/ifcfg-' . $_name;
-    $_ipaddr = '0.0.0.0' if ($_ipaddr eq '');
-    $_ipaddr = '' if (!($_ipaddr =~ /\d+\.\d+\.\d+\.\d+/));
+    my ($self, %args) = @_;
+
+    $self->reveal_myself;
+    $args{_ipaddr} //= '0.0.0.0';
+    $args{_name} //= '';
+    $args{_bootproto} //= 'dhcp';
+    $args{_startmode} //= 'auto';
+    $args{_zone} //= '';
+    $args{_bridge_type} //= 'master';
+    $args{_bridge_port} //= '';
+    $args{_bridge_stp} //= 'off';
+    $args{_bridge_forwarddelay} //= '15';
+    croak("Interface name must be given otherwise network bridge device config can not be generated.") if ($args{_name} eq '');
+
+    $args{_ipaddr} = '0.0.0.0' if ($args{_ipaddr} eq '');
+    $args{_ipaddr} = '' if (!($args{_ipaddr} =~ /\d+\.\d+\.\d+\.\d+/));
+    if (is_transactional) {
```

Comment:
This seems odd to me. There can be transactional systems with wicked (e.g. SLE Micro 5.2) and non-transactional systems with NetworkManager.

Reply:
It is hard to know everything in advance, but the logic here is clear. So the "if" condition can be tailored in the future to have more precise control over time.

Reply:
`lib/mm_network.pm` already exports a function `is_networkmanager`, please use that instead of something which works only by accident.

Reply:
Done